src/bid/bid.rs

```diff
@@ -60,6 +59,81 @@ impl Borrow<u64> for Bid {
     }
 }
 
+impl PartialEq for Bid {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash().eq(&other.hash())
+    }
+}
+
+// This needs to be between braces since const fn calls passed as const_generics
+// params aren't perfectly supported yet.
+impl Serializable<{ Bid::serialized_size() }> for Bid {
+    type Error = BlindBidError;
```

Comment:
After we used the trait for a while, I'm still not convinced this is the way to go; but I want to discuss as soon as we have time (maybe in the Q/A): basically doing that we potentially have multiple errors returned by `from_bytes` implementation and it's hard to tell from a consumer library. Often is easier just return a single error where the conversion failed.
But I'm not sure, I'd like to discuss with the rest of you guys.